### PR TITLE
Nextjs-Auth0 Integration

### PR DESCRIPTION
**Pull request by Stoner-Bytelion**
### What I changed
* Integrated nextjs-auth0 and confirmed working with vercel
* Set and tested index page as a protected page
* Included vercel exception to logging int
* Updated out of date packages
* Updated alias import usage

### CODE REVIEW

**Comment by Stoner-Bytelion** [comment]
* Updated readme to include authentication flow of the application

File: README.md [hunk 1]
```diff
@@ -35,31 +35,19 @@ npm run custom
 * `.eslintrc`: main linting configuration
 * `next.config.js`: contains environment variables and plugin configurations for nextjs
 
-## Algolia
```
Comment (m-torin) [inline README.md:38]: Did you take Algolia out?

File: .eslintrc [hunk 2]
```diff
@@ -25,18 +27,28 @@
     "import/no-extraneous-dependencies": 0,
     "import/prefer-default-export": "off",
     "max-len": "off",
+    "no-param-reassign": ["error", { "props": false }],
```
Comment (m-torin) [inline .eslintrc:30]: What's the use case why this is needed?

File: package.json [hunk 3]
```diff
@@ -8,37 +8,32 @@
     "start": "next start",
     "lint": "eslint . --ext .js --fix"
   },
-  "husky": {
-    "hooks": {
-      "pre-commit": "npm run lint"
-    }
-  },
+  "license": "MIT",
```
Comment (m-torin) [inline package.json:11]: Please change to `Unlicensed`

File: package.json [hunk 4]
```diff
@@ -8,37 +8,32 @@
     "start": "next start",
     "lint": "eslint . --ext .js --fix"
   },
-  "husky": {
-    "hooks": {
-      "pre-commit": "npm run lint"
-    }
-  },
```
Comment (m-torin) [inline package.json:15]: Lol; an explanation if you please sir. :)

File: src/components/auth/login-redirect.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import Router from 'next/router';
+import React, { Component } from 'react';
+
+import Layout from '@/components/layout';
+import createLoginUrl from '@/lib/url-helper';
+
+export default class RedirectToLogin extends Component {
```
Comment (m-torin) [inline src/components/auth/login-redirect.js:7]: Why does this need to be a class? We could dynamically import the module instead.

If this is an Auth0 file, please note it at the top and add a TO-DO.

File: src/components/auth/with-auth.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable react/prop-types */
+
+import React, { Component } from 'react';
+
+import auth0 from '@/lib/auth0';
+import { fetchUser } from '@/lib/user';
+import createLoginUrl from '@/lib/url-helper';
+import RedirectToLogin from './login-redirect';
```
Comment (m-torin) [inline src/components/auth/with-auth.js:16]: Please use the `@`.

File: src/components/auth/with-auth.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+          });
+          ctx.res.end();
+        }
+
+        return {
+          session: {
+            user: {
+              name: 'Exception',
+            },
+            createdAt: 1592859746557,
```
Comment (m-torin) [inline src/components/auth/with-auth.js:42]: Why is the EPOCH hardcoded?

File: src/lib/user.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
+/**
```
Comment (m-torin) [inline src/lib/user.js:1]: Please add a note that this is an Auth0 file

File: src/pages/_app.js [hunk 9]
```diff
@@ -2,12 +2,11 @@
 
 import React from 'react';
 import App from 'next/app';
-import { Provider } from 'next-auth/client';
 import whyDidYouRender from '@welldone-software/why-did-you-render';
-import AppContextProvider from 'contexts';
+import AppContextProvider from '@/contexts/index';
```
Comment (m-torin) [inline src/pages/_app.js:6]: Index is implied isn't it?

File: src/pages/_app.js [hunk 10]
```diff
@@ -2,12 +2,11 @@
 
 import React from 'react';
 import App from 'next/app';
-import { Provider } from 'next-auth/client';
```
Comment (m-torin) [inline src/pages/_app.js:5]: Why are we removing the provider from the main wrapper?

**Review by m-torin — COMMENTED** [review]
Why did we remove Algolia?

File: package.json (hunk 4 above)
Comment (Stoner-Bytelion) [reply package.json:15]: A few times it's made pulling a little prone to friction but if you like it I'll keep it in there!

File: src/pages/_app.js (hunk 9 above)
Comment (Stoner-Bytelion) [reply src/pages/_app.js:6]: oddly enough it throws a "Can't resolve" error without index

File: src/pages/_app.js (hunk 10 above)
Comment (Stoner-Bytelion) [reply src/pages/_app.js:5]: This is from the other next-auth library that I thought would serve us better at the time

**Pull request merged**